What's new article (#28029)
dotnet/docs: docs/whats-new/dotnet-docs-2022-01-01.md

---

> title: ".NET docs: What's new for January 1, 2022 - January 31, 2022"

Reviewer: ```suggestion
title: ".NET docs: What's new for January 2022
```

> description: "What's new in the .NET docs for January 1, 2022 - January 31, 2022."

Reviewer: ```suggestion
description: "What's new in the .NET docs for January 2022
```

ms.date: 02/01/2022
---

> # .NET docs: What's new for January 1, 2022 - January 31, 2022

Reviewer 2: ```suggestion
# .NET docs: What's new for January 2022
```

> Welcome to what's new in the .NET docs from January 1, 2022 through January 31, 2022. This article lists some of the major changes to docs during this period.

Reviewer: ```suggestion
Welcome to what's new in the .NET docs for January 2022. This article lists some of the major changes to docs during this period.
```

- [Deserialize Version type with leading or trailing whitespace](../core/compatibility/serialization/7.0/deserialize-version-with-whitespace.md)
- [Set timestamp on read-only file on Windows](../core/compatibility/core-libraries/6.0/set-timestamp-readonly-file.md)
- [.NET implementations](../fundamentals/implementations.md)
- [Ecma standards](../fundamentals/standards.md)
- [Breaking changes in .NET 7](../core/compatibility/7.0.md)
- [Validate CompressionLevel for BrotliStream](../core/compatibility/core-libraries/7.0/compressionlevel-validation.md)
- [Default serialization format for TimeSpan](../core/compatibility/serialization/6.0/timespan-serialization-format.md)
- [IL3003: 'RequiresAssemblyFilesAttribute' annotations must match across all interface implementations or overrides.](../fundamentals/code-analysis/quality-rules/il3003.md)

### Updated articles

- [Upgrade a Windows Forms App to .NET 6 with the .NET Upgrade Assistant](../core/porting/upgrade-assistant-winforms-framework.md) - Upgrade assistant updates
- [Upgrade a WPF App to .NET 6 with the .NET Upgrade Assistant](../core/porting/upgrade-assistant-wpf-framework.md) - Upgrade assistant updates

> ## Community contributors

Reviewer 3: Add an announcement of updating the source for the C# specification:

```suggestion
## C# language

### Updated articles

- [ECMA C# (draft) standard](csharp/language-reference/language-specification/readme) - Publish the C# draft standard instead of the Microsoft C# draft specification.

## Community contributors
```
